src/momento/internal/synchronous/_scs_control_client.py

```diff
@@ -1,70 +1,124 @@
+            _validate_cache_name(cache_name)
             request = _DeleteCacheRequest()
             request.cache_name = cache_name
-            self._get_stub().DeleteCache(request, timeout=_DEADLINE_SECONDS)
-            return DeleteCacheResponse()
+            self._getStub().DeleteCache(request, timeout=_DEADLINE_SECONDS)
         except Exception as e:
-            logs.debug(f"Failed to delete cache: {cache_name} with exception:{e}")
-            raise _cache_service_errors_converter.convert(e) from None
+            self._logger.debug("Failed to delete cache: %s with exception: %s", cache_name, e)
+            return DeleteCacheResponse.Error(convert_error(e))
+        return DeleteCacheResponse.Success()
 
-    def list_caches(self, next_token: Optional[str] = None) -> ListCachesResponse:
+    def list_caches(self, next_token: Optional[str] = None) -> ListCachesResponseBase:
         try:
             list_caches_request = _ListCachesRequest()
             list_caches_request.next_token = next_token if next_token is not None else ""
-            return ListCachesResponse.from_grpc_response(
-                self._get_stub().ListCaches(list_caches_request, timeout=_DEADLINE_SECONDS)
+            response = self._getStub().ListCaches(list_caches_request, timeout=_DEADLINE_SECONDS)
+            return ListCachesResponse.Success.from_grpc_response(response)
+        except Exception as e:
+            return ListCachesResponse.Error(convert_error(e))
+
+    def create_signing_key(self, ttl: timedelta, endpoint: str) -> CreateSigningKeyResponse:
+        try:
+            _validate_ttl(ttl)
+            ttl_minutes = round(ttl.total_seconds() / 60)
+            self._logger.info(f"Creating signing key with ttl (in minutes): {ttl_minutes}")
+            create_signing_key_request = _CreateSigningKeyRequest()
+            create_signing_key_request.ttl_minutes = ttl_minutes
+            return CreateSigningKeyResponse.from_grpc_response(
+                self._getStub().CreateSigningKey(create_signing_key_request, timeout=_DEADLINE_SECONDS),
+                endpoint,
+            )
+        except Exception as e:
+            self._logger.warning(f"Failed to create signing key with exception: {e}")
+            raise convert_error(e)
+
+    def revoke_signing_key(self, key_id: str) -> RevokeSigningKeyResponse:
+        try:
+            self._logger.info(f"Revoking signing key with key_id {key_id}")
+            request = _RevokeSigningKeyRequest()
+            request.key_id = key_id
+            self._getStub().RevokeSigningKey(request, timeout=_DEADLINE_SECONDS)
+            return RevokeSigningKeyResponse()
+        except Exception as e:
+            self._logger.warning(f"Failed to revoke signing key with key_id {key_id} exception: {e}")
+            raise convert_error(e)
+
+    def list_signing_keys(self, endpoint: str, next_token: Optional[str] = None) -> ListSigningKeysResponse:
```

Comment:
I don't think we should introduce the signing key methods in the synchronous client at this point unless somebody specifically asks for them.

Reply:
Disagree here. I think we should have API parity between the async and sync clients, given it's the same language and codebase. A user could decide to switch from the async to sync or vice versa and would expect the top level API to be the same.

Reply:
They were only in the async client before this work. I guess it doesn't hurt anything, but I'm hesitant to increase our surface area unless there's a real need for it.

Reply:
I think the API parity argument makes sense given Saturn's experience but we definitely need to have the conversation soon (**definitely** pre-`1.0`) about whether we want to pull these APIs out to a separate package.  I will make a note to discuss that with Meera.